### DISCUSSION TITLE
which type should be use with Mapped type for postgresql specifical dialects?

### DISCUSSION THREAD

**Discussion by wangxin688** [sqlalchemy/sqlalchemy#9437] [Usage Questions] [answered]
there are some db-related types like `JSONB`, `JSON`, `HSTORE`, `ARRAY` type， how can I represent these types use `Mapped` and `mapped_colunm`, I don't know if this is the correct way to mark the data type.
```python
class AlertMixin:
    id: Mapped[uuid.UUID] = mapped_column(primary_key=True, server_default=GUID_GEN)
    status: Mapped[Literal["Firing", "Resolved"]] = mapped_column(ENUM("Firing", "Resolved", name="alert_status"))
    started_at: Mapped[datetime] = mapped_column(DateTime(timezone=True), default=func.now(), index=True)
    updated_at: Mapped[datetime] = mapped_column(DateTime(timezone=True), onupdate=func.now())
    duration: Mapped[int]
    mac_address: Mapped[MACADDR] = mapped_column(MACADDR, comment="ap mac address")
    serial_number: Mapped[str] = mapped_column(String, comment="serial number")
    ip_address: Mapped[INET] = mapped_column(INET, comment="ap ip address")

```

**Comment by CaselIT** [comment, marked as answer]
Hi,

No, that is not. Mapped requires the python type in the annotation, not the sql one.

I think inet and macaddr are both converted to string in python, so use str

**Reply by wangxin688** [reply]
So that is to say, we need declare the real db type in mapped_column. Thanks

**Reply by CaselIT** [reply]
yes, if that's a common thing you may try using annotated to create "annotated" python types that are resolved by sqlalchemy to the proper db type.
See https://docs.sqlalchemy.org/en/20/orm/declarative_tables.html#mapping-multiple-type-configurations-to-python-types
